File: DK/pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

//https://cs50.harvard.edu/x/2020/psets/2/readability/

int main(void)
{
    //get user input text as string
    string text = get_string("Text: ");
    //determine length of input text
    int length = strlen(text);
    float letterCount = 0;
    float wordCount = 1;
    float sentenceCount = 0;

    //iterate over input string to determine letterCount, wordCount and sentenceCount
    for (int i = 0; i < length; i++)
    {
        if ((text[i] <= 'z' && text[i] >= 'a') || (text[i] <= 'Z' && text[i] >= 'A'))
        {
            letterCount ++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentenceCount ++;
        }
        else if (text[i] == ' ')
        {
            wordCount ++;
        }
    }

    //determine average number of letters per 100 words
    float L = letterCount / wordCount * 100;

    //determine average number of sentences per 100 words
    float S = sentenceCount / wordCount * 100;

    //calculate coleman-liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    /*
    printf("%d letters\n", (int) letterCount);
    printf("%d words\n", (int) wordCount);
    printf("%d sentences\n", (int) sentenceCount);
    printf("%f\n", index);
    */

    //print grade
    if (grade >= 1 && grade <= 16)
    {
        printf("Grade %d\n", grade);
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
}

File: DK/pset6/readability/readability.py
# https://cs50.harvard.edu/x/2020/psets/6/readability/


# Initialise text and prompt user for input
text = None
while text == None:
    text = input("Text: ")

# Initialise variables
letterCount = 0
wordCount = 1
sentenceCount = 0

# Iterate through every character in input string
for char in text:
    # Case where character is a letter
    if char.isalpha():
        letterCount += 1
    # Case where character is a space
    elif char == " ":
        wordCount += 1
    # Case where character is a sentence indicator
    elif char == '.' or char == '?' or char == '!':
        sentenceCount += 1
        
# Determine average number of letters per 100 words
L = letterCount / wordCount * 100

# Determine averge number of sentences per 100 words
S = sentenceCount / wordCount * 100

# Calculate coleman-liau index
index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)

# Print grade
if grade > 0 and grade < 17:
    print(f"Grade {grade}")
elif grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
